block_qr.php

```diff
@@ -230,8 +239,17 @@ public function get_content() {
             $fullview = true;
         }
 
-        $urlshort = urlencode($qrcodelink);
+        // Short link url in admin settings.
+        $configshortlink = get_config('block_qr', 'configshortlink');
 
+        // Short link service.
+        if (empty($configshortlink)) {
+            $urlshort = null;
+        } else {
+            $urlshort = str_replace('SHORTLINK', urlencode($qrcodelink), $configshortlink);
```

Comment:
```suggestion
            $urlshort = urlencode(str_replace('SHORTLINK', $qrcodelink, $configshortlink));
```

Reply:
Unfortunately, this now messes up the $qrcodelink as a whole, so the QR code has the wrong link.

So I'd suggest to go for:
```
            $encodedqrcodelink = urlencode($qrcodelink);
            $urlshort = str_replace('SHORTLINK', $encodedqrcodelink, $configshortlink);
```
or maybe even use what you had initially. Sorry for the noise :see_no_evil: